server/webapp/WEB-INF/rails.new/config/routes.rb

```diff
@@ -228,6 +228,9 @@
         patch :update, on: :member
       end
 
+      get 'stages/:pipeline_name/:pipeline_counter/:stage_name/:stage_counter/instance' => 'stages#show', constraints: {pipeline_name: PIPELINE_NAME_FORMAT, pipeline_counter: PIPELINE_COUNTER_FORMAT, stage_name: STAGE_NAME_FORMAT, stage_counter: STAGE_COUNTER_FORMAT}, as: :stage_instance_by_counter_api
+      get 'stages/:pipeline_name/:stage_name/history' => 'stages#history', constraints: {pipeline_name: PIPELINE_NAME_FORMAT, stage_name: STAGE_NAME_FORMAT}, as: :stage_history_api
+
```

Comment:
Have you given it a though about how all the new URLs (pipeline, job, stage) look like? It feels like this is a replica of the old url format. 

Is `stages/:pipeline_name/:pipeline_counter/:stage_name/:stage_counter` and `stages/:pipeline_name/:stage_name` not enough?